PusherServer/BatchEvent.cs

```diff
@@ -0,0 +1,28 @@
+﻿namespace PusherServer
+{
+    /// <summary>
+    /// Represents an event that is part of a batch trigger
+    /// </summary>
+    public class BatchEvent
+    {
+        /// <summary>
+        /// The name of the event
+        /// </summary>
+        public string name { get; set; }
+
+        /// <summary>
+        /// The event data
+        /// </summary>
+        public string data { get; set; }
```

Comment:
I'd suggest we make this an `object` from a public API point of view. We should also consider making the properties upper-case to adhere to the C# convention.